Subject: Disambiguate PYTHON_FOR_BUILD variable

On DEL/DBL, previously we relied on defining PYTHON_FOR_BUILD as a way
to pass the path to the "host" Python interpreter. With the upgrade
to Python 2.7.18 though, PYTHON_FOR_BUILD now also tries to include
a handful of environment variables, including (crucially)
_PYTHON_HOST_PLATFORM, which serves the same purpose as our
CROSS_COMPILING variable.

Since it would be fragile for us to copy the PYTHON_FOR_BUILD variable
definition out to our package Makefile, we will instead add the ability
to define HOST_PYTHON_FOR_BUILD, in which case we will slot that executable
into the PYTHON_FOR_BUILD variable. This way, we can take advantage of the
improved cross-compilation support from upstream (and the Buildroot patches),
without making our lives harder.

Signed-off-by: Mike Wadsten <mike.wadsten@digi.com>

--- a/configure.ac
+++ b/configure.ac
@@ -25,7 +25,7 @@
 if test "$cross_compiling" = yes; then
     AC_MSG_CHECKING([for python interpreter for cross build])
     if test -z "$PYTHON_FOR_BUILD"; then
-        for interp in python$PACKAGE_VERSION python2 python; do
+        for interp in "$HOST_PYTHON_FOR_BUILD" python$PACKAGE_VERSION python2 python; do
 	    which $interp >/dev/null 2>&1 || continue
 	    if $interp -c 'import sys;sys.exit(not (sys.version_info@<:@:2@:>@ >= (2,7) and sys.version_info@<:@0@:>@ < 3))'; then
 	        break
